crates/wasmtime/src/runtime/module.rs

```diff
@@ -238,11 +249,15 @@ impl Module {
     /// ```
     #[cfg(any(feature = "cranelift", feature = "winch"))]
     #[cfg_attr(docsrs, doc(cfg(any(feature = "cranelift", feature = "winch"))))]
-    pub fn new(engine: &Engine, bytes: impl AsRef<[u8]>) -> Result<Module> {
+    pub fn new(
+        engine: &Engine,
+        bytes: impl AsRef<[u8]>,
+        wasm_path: Option<PathBuf>,
```

Comment:
@peterhuene Hi, tagging you as the bot marked you as reviewer, hope that's ok?  You'll see I've added a parameter to the constructor here so that I can load the `.dwp` file at the same time as loading the `.wasm` (if loading from file).  This means there are a zillion tests to update, which is fine with me, but before I do it, do you have any comment about this aspect of this PR?

Reply:
If possible I'd recommend avoiding changing the default constructor of `Module` since, as you've seen, there's a lot of callers. Additionally dealing with split-dwarf-things feels somewhat niche-ish where embedders should all be able to handle it but it shouldn't necessarily be front-and-center of the embedding API.

If you're up for it what I might recommend for something like this is perhaps a `ModuleBuilder` type, for example:

```rust
impl Module {
    pub fn builder() -> ModuleBuilder;
}

impl ModuleBuilder {
    fn allow_wat(&mut self, allow: bool) -> &mut Self; // example configuration method
    fn dwarf_package(&mut self, bytes: &[u8]) -> &mut Self; // explicitly specify dwarf
    fn dwarf_package_file(&mut self, path: impl AsRef<Path>) -> Result<&mut Self>; // explicitly specify path
    
    // finish the compilation by reading wasm from `path`
    //
    // auto-load `path.dwp` if `dwarf_package` files weren't called
    fn compile_path(&mut self, path: impl AsRef<Path>) -> Result<Module>;

    // finish compilation by using the bytes provided as wasm
    fn compile(&mut self, wasm: &[u8]) -> Result<Module>;
}
```

or something along those lines. That'll keep `Module::new` working as-is. `Module::from_file` would probe for a dwarf package (perhaps with a knob in `Config` to turn that off), and users with in-memory wasm and dwarf packages can use `ModuleBuilder`

Reply:
Sounds good to me, thanks!   Will get on it.

Reply:
I've taken your outline for `ModuleBuilder` and have used it as the sole mechanism for creating a module with split dwarf.  So I've left `Module::from_file` and `new` as is.  To be honest I did that, then re-read your comment, so if you think it would be better if `from_file` understood to read dwarf packages automatically, let me know.